ui/src/components/settings/sub_pages/profile/mod.rs

```diff
@@ -458,6 +494,50 @@ pub fn ProfileSettings(cx: Scope) -> Element {
                         }
                     },
                 },
+                SettingSection {
+                    section_label: get_local_text("settings-profile.recovery-seed"),
+                    section_description: get_local_text("settings-profile.recovery-seed-description"),
+                    Button {
+                        text: if seed_phrase.as_ref().is_none() { get_local_text("settings-profile.reveal-seed") } else { get_local_text("settings-profile.hide-seed") },
+                        aria_label: "reveal-recovery-seed-button".into(),
+                        appearance: Appearance::Danger,
+                        icon: if seed_phrase.as_ref().is_none() { Icon::Eye } else { Icon::EyeSlash },
+                        onpress: move |_| {
+                            if seed_phrase.is_some() {
+                                seed_phrase.set(None);
+                            } else {
+                                seed_words_ch.send(());
+                            }
+                        }
+                    }
+                }
+                if let Some(phrase) = seed_phrase.as_ref() {
+                    let words = phrase.split_whitespace().collect::<Vec<&str>>();
+                    render!(
+                        SettingSectionSimple {
+                            div {
+                                class: "seed-words",
+                                words.chunks_exact(2).enumerate().map(|(idx, vals)| rsx! {
+                                    div {
+                                        class: "row",
+                                        div {
+                                            class: "col",
+                    
+                                            span { class: "num", ((idx * 2) + 1).to_string() },
+                                            span { class: "val", vals.get(0).cloned().unwrap_or_default() }
```

Comment:
Resolving a future linting issue

```suggestion
                                            span { class: "val", vals.first().cloned().unwrap_or_default() }
```